test/Libraries/CoreNodesTests/CoreNodeModelsTests/SelectionTests.cs

```diff
@@ -0,0 +1,222 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Xml;
+using CoreNodeModels;
+using CoreNodeModels.Properties;
+using Dynamo;
+using Dynamo.Graph;
+using Dynamo.Graph.Nodes;
+using NUnit.Framework;
+using Moq;
+using TestUINodes;
+
+namespace DSCoreNodesTests
+{
+    [TestFixture]
+    public class SelectionTests : DynamoModelTestBase
+    {
+        private SelectionConcrete selection;
+        private NodeModel testNode;
+        private Mock<IModelSelectionHelper<ModelBase>> selectionHelperMock;
+
+        [SetUp]
+        public void TestsSetup()
+        {
+            selectionHelperMock = new Mock<IModelSelectionHelper<ModelBase>>(MockBehavior.Strict);
+            selection = new SelectionConcrete(SelectionType.Many, SelectionObjectType.Element, "testMessage", "testPrefix", selectionHelperMock.Object);
+            selection.Name = "selectionTestName";
+            testNode = new DummyNode();
```

Comment:
Can we test this with some other node? I think you used `CodeBlockNode` before. Will that work now? Selecting a DummyNode may not be all that useful to test.

Reply:
Yes it should work, I'll change it and update the pr